src/Umbraco.Core/Services/MemberService.cs

```diff
@@ -1199,7 +1198,7 @@ public void DissociateRoles(int[] memberIds, string[] roleNames)
 
         private IMemberType FindMemberTypeByAlias(string memberTypeAlias)

```

Comment:
I don't think we should be committing the unit of work here. If we for some reason should be (this is related to my other comment as well where I'm asking the same) .. we should not be running this in a readonly unit of work.